cmd/tsgolint/headless.go

```diff
@@ -154,47 +160,56 @@ func runHeadless(args []string) int {
 		return 1
 	}
 
-	var config headlessConfig
+	var payload headlessPayload
 
-	if err := json.Unmarshal(configRaw, &config); err != nil {
-		writeErrorMessage(fmt.Sprintf("error parsing config: %v", err))
+	if err := json.Unmarshal(configRaw, &payload); err != nil {
+		writeErrorMessage(fmt.Sprintf("error parsing payload: %v", err))
 		return 1
 	}
-	if len(config.Files) == 0 {
-		writeErrorMessage("no files specified in config")
+	if len(payload.Configs) == 0 {
+		writeErrorMessage("invalid payload format: 'configs' field is required and must not be empty")
 		return 1
 	}
 
-	fileConfigs := make(map[string]headlessConfigForFile, len(config.Files))
 	workload := linter.Workload{
 		Programs:       make(map[string][]string),
 		UnmatchedFiles: []string{},
 	}
 
 	if logLevel == utils.LogLevelDebug {
-		log.Printf("Starting to assign files to programs. Total files: %d", len(config.Files))
+		totalFileCount := 0
+		for _, config := range payload.Configs {
+			totalFileCount += len(config.FilePaths)
+		}
+		log.Printf("Starting to assign files to programs. Total files: %d", totalFileCount)
 	}
 
 	tsConfigResolver := utils.NewTsConfigResolver(fs, cwd)
 
-	for idx, fileConfig := range config.Files {
-		if logLevel == utils.LogLevelDebug {
-			log.Printf("[%d/%d] Processing file: %s", idx+1, len(config.Files), fileConfig.FilePath)
-		}
+	var fileConfigs = make(map[string]*[]headlessRule, len(payload.Configs))
```

Comment:
I think the len passed to `make` should be the `totalFileCount`, rather than the number of diff configs